crates/autopilot/src/protocol/orderbook.rs

```diff
@@ -14,13 +17,17 @@ impl Orderbook {
     }
 
     /// Retrieves the current auction.
-    pub async fn auction(&self) -> reqwest::Result<AuctionWithId> {
+    pub async fn auction(&self) -> reqwest::Result<domain::AuctionWithId> {
         self.client
             .get(shared::url::join(&self.url, "api/v1/auction"))
             .send()
             .await?
             .error_for_status()?
-            .json()
+            .json::<dto::AuctionWithId>()
```

Comment:
Can be done later, but `self.client.get` should probably also be abstracted in an infra module that directly returns the domain auction

Reply:
👍